lib/m2ts/caption-stream.js

```diff
@@ -847,16 +893,23 @@ Cea608Stream.prototype.shiftRowsUp_ = function() {
   for (i = 0; i < this.topRow_; i++) {
     this.displayed_[i] = '';
   }
+  for (i = this.row_ + 1; i < BOTTOM_ROW + 1; i++) {
+    this.displayed_[i] = '';
+  }
   // shift displayed rows up
-  for (i = this.topRow_; i < BOTTOM_ROW; i++) {
+  for (i = this.topRow_; i < this.row_; i++) {
     this.displayed_[i] = this.displayed_[i + 1];
   }
   // clear out the bottom row
-  this.displayed_[BOTTOM_ROW] = '';
+  this.displayed_[this.row_] = '';
 };
 
-// paintOn mode is not implemented
-Cea608Stream.prototype.paintOn = function() {};
+Cea608Stream.prototype.paintOn = function(pts, text) {
+  var baseRow = this.displayed_[this.row_];
```

Comment:
The way formatting is being done is still an issue (just generally). Also, if that's the case, the code needs to be something more like
```javascript
  baseRow = baseRow.slice(0, this.column_) + text + baseRow.slice(this.column_ + text.length);
```

Otherwise, your text might end up in the wrong position.

I have an idea to switch the formatting implementation to injecting non-printable characters representing formatting and then only replacing for the HTML tags when the captions are dispatched. That would allow for arbitrary column positionining, which is necessary for spec-compliant paint-on. In any case, that's not in this PR.

Reply:
I agree that replacing the formatting right before dispatching captions makes the most sense.

What kind of non-printable characters did you have in mind?

Reply:
0x10 through 0x1f. Basically `<format code> & 0xf`

Reply:
Hmm, I would be concerned that some of these are already used (aren't 0x14 through 0x1f used for standard characters?) but we don't need to address that in this PR.

How do you feel about creating an issue to collect ideas on how to fix the formatting and keep track of it as a separate issue?

Reply:
Not to my knowledge no, I think the only characters < 0x20 that are used are the cursor-moving characters 0x08 through 0x0d.

But yeah, I can open a separate issue.